### PR TITLE
Use github actions workflow to build and push to dockerhub

### PR DESCRIPTION
**Pull request by edwardrf**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```
Comment (lionello) [inline .github/workflows/build.yml:16]: I'm surprised this is needed

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+name: Build and Push to Dockerhub
+
+on:
+  push:
+    branches:
+      - master
```
Comment (lionello) [inline .github/workflows/build.yml:6]: should we do a build, no push, on PRs?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      - name: Set version to short commit SHA
+        run: |
+          shortSha=$(git rev-parse --short ${{ github.sha }})
+          echo "VERSION=$shortSha" >> $GITHUB_ENV
+
+      - name: Set up Go environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
+          cache: false # Avoid warning: https://github.com/actions/setup-go/issues/403
```
Comment (lionello) [inline .github/workflows/build.yml:33]: That issue is for v4, is this still present in v5? Caching sounds like something that's good to have (and we use this action in other repos without issues?)

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+      - name: Set version to short commit SHA
+        run: |
+          shortSha=$(git rev-parse --short ${{ github.sha }})
+          echo "VERSION=$shortSha" >> $GITHUB_ENV
+
+      - name: Set up Go environment
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.21'
```
Comment (lionello) [inline .github/workflows/build.yml:32]: you can tell the action to get the go-version from `go.mod`

File: .github/workflows/build.yml (hunk 1 above)
Comment (edwardrf) [reply .github/workflows/build.yml]: Yeah, me too until I realised [github does not have arm64 runners](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners), so we still have to build the arm image on a x86 setup, which relies on qemu.

File: .github/workflows/build.yml (hunk 2 above)
Comment (edwardrf) [reply .github/workflows/build.yml:6]: Added a build only workflow for PRs

File: .github/workflows/build.yml (hunk 3 above)
Comment (edwardrf) [reply .github/workflows/build.yml:24]: Yep, tested, it is still there, and according to the linked issue, someone tested, it is actually faster with `cache: false` somehow...

**Pull request merged**